src/gpu/generic/sycl/ref_pooling.cpp

```diff
@@ -65,18 +65,55 @@ status_t ref_pooling_fwd_t::pd_t::init_conf() {
     conf_.DW = KDW(); //K:kernel D:Dilation W:Weight
 
     conf_.post_ops = sycl_post_ops_t(attr(), dst_md());
+
+    src_zero_dim_ = memory_desc_wrapper(src_md()).size() == 0;
+    dst_zero_dim_ = memory_desc_wrapper(dst_md()).size() == 0;
+
     return status::success;
 }
 
 status_t ref_pooling_fwd_t::init(impl::engine_t *engine) {
-    const auto kid = ::sycl::get_kernel_id<pooling_fwd_kernel_vec_t>();
-    return create_kernel(engine, kid, &kernel_);
+    if (pd()->src_zero_dim_ && !pd()->dst_zero_dim_) {
+        const auto fill_kid = ::sycl::get_kernel_id<pooling_fwd_kernel_vec_t>();
+        CHECK(create_kernel(engine, fill_kid, &fill_kernel_));
+    } else {
+        const auto kid = ::sycl::get_kernel_id<pooling_fwd_kernel_vec_t>();
+        CHECK(create_kernel(engine, kid, &kernel_));
+    }
+
+    return status::success;
 }
 
 status_t ref_pooling_fwd_t::execute_forward(const exec_ctx_t &ctx) const {
-    // XXX: Add support for 0-dim src
-    for (auto &md : {pd()->src_md(), pd()->dst_md()}) {
-        if (memory_desc_wrapper(md).size() == 0) return status::success;
+    if (pd()->dst_zero_dim_) return status::success;
+    if (pd()->src_zero_dim_ && !pd()->dst_zero_dim_) {
+        auto dst_mdw = memory_desc_wrapper(pd()->dst_md());
+        float fill_val = 0.f;
+
+#define CASE(DT) \
+    case dnnl_##DT: \
+        fill_val = std::numeric_limits< \
+                prec_traits<data_type::DT>::type>::lowest(); \
+        break;
+
+        switch (dst_mdw.data_type()) {
+            CASE(f16)
+            CASE(bf16)
+            CASE(f32)
+            CASE(s8)
+            CASE(u8)
+            default:
+                assert(!"unhandled datatype");
+                return status::unimplemented;
```

Comment:
This error should be reported at `pd_t::init()` level. Here the only thing possible is runtime_error which shouldn't be reachable by the user.

Reply:
There is already a check for the supported datatypes in the `pd::init()` function. I mainly added this default case here in case we ever add more datatypes to pooling and forget to update this switch. 
Changed it to `runtime_error`, thanks for the suggestion.